### PR TITLE
Strato Rollup Plugin

### PR DESCRIPTION
**Pull request by 3Nigma**

### CODE REVIEW

File: lib.docs/src/strato/rollup.config.js [hunk 1]
```diff
@@ -44,29 +30,14 @@ export default async function getConfig() {
...
-      alias({
-        entries: [
-          { find: 'dotenv', replacement: getPathOf("./polyfills/dotenv.js") },
-          { find: /.*SolidityCompiler.*/, replacement: getPathOf("./polyfills/SolidityCompiler.js") },
-          { find: /.*StratoLogger.*/, replacement: getPathOf("./polyfills/StratoLogger.js") },
-
-          // Alias-ing browser specific implementations
-          { find: /(.*encoding\/Hex.*)/, replacement: '$1.native' }
-        ],
+      strato({
```
Comment (3Nigma) [inline lib.docs/src/strato/rollup.config.js:33]: This is how using the plugin will look like. We'll have a separate demo-repo to highlight this better.

**Review by vla-dev — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -32,8 +28,8 @@ Strato already supports a lot of stuff:
 * [x] Create a Hedera account
 * [x] Ready to be plugged into a web3 wallet (_when such will exist_)
 * [x] End to end tested sourcing multiple contracts for the test-base from places such as [solidity-by-example](https://solidity-by-example.org/) and the [hedera-sdk-js repo](https://github.com/hashgraph/hedera-sdk-js/tree/main/examples)
+* [x] Bunddling via [Rollup](https://rollupjs.org/) with the aid of a custom defined plugin
```
Comment (victorholo) [inline README.md:31]: There is a typo here.

File: lib.docs/markdown/guides/bundling.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
+---
+id: bundling
+title: Bundling
+---
+
+Currently we offer support for bundling strato via [rollup](https://rollupjs.org/) with support for other bundlers [being scheduled](https://github.com/buidler-labs/hedera-strato-js/issues/26), yet not commited.
+
+### General considerations
+Strato is delivered in both `es-module` and `common-js` formats. The challenge with bundling the library boils down to whether or not you wish to use the `SolidityCompiler` function in-browser. If you wish in-browser compilation, a web-worker is provided which fetches the appropriate solidity-compiler binary before carring out the compilation itself via calling any of the `Contract.newFrom`/`Contract.allFrom` familiy of functions.
```
Comment (victorholo) [inline lib.docs/markdown/guides/bundling.md:9]: "familiy" of functions - typo as well :)

File: lib/bundler/rollup/plugin.ts [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  environment?: NodeJS.ProcessEnv,
+  includeCompiler?: boolean,
+  sourceMap?: boolean, 
+};
+
+const CONTRACT_REGISTRY_ID = '\0hedera-strato:ContractRegistry';
+const CONTRACTS_IN_FILE_STORAGE_ID = '\0hedera-strato:ContractsInFileStorage';
+const SOLIDITY_COMPILER_ID = '\0hedera-strato:SolidityCompiler';
+
+export default function strato(options: StratoRollupOptions = {}) {
```
Comment (victorholo) [inline lib/bundler/rollup/plugin.ts:23]: 👏👏🤟

**Review by victorholo — APPROVED** [review]

**Pull request merged**